src/OpenDiffix.Core/Expression.fs

```diff
@@ -190,13 +190,13 @@ let rec evaluateScalarFunction fn args =
     | true, r -> Real r
     | false, _ -> Null
   | Cast, [ String s; String "boolean" ] ->
-    match s.ToLower() with
+    match s.Trim().ToLower() with
     | "true"
     | "1" -> Boolean true
     | "false"
     | "0" -> Boolean false
-    | "" -> Null
-    | _ -> failwith "Input value is not a valid boolean string."
+    | ""
+    | _ -> Null
```

Comment:
Can be merged to only `_ -> Null`

Reply:
heh, I knew you'd suggest that. I kept separate since cases are different, but it's fine merged too.